libretro-rs/src/core_macro.rs

```diff
@@ -37,12 +37,10 @@ macro_rules! libretro_core {
 
       #[no_mangle]
       extern "C" fn retro_init() {
-        instance_mut(|instance| instance.on_init())
       }
 
       #[no_mangle]
       extern "C" fn retro_deinit() {
-        instance_mut(|instance| instance.on_deinit())
       }
```

Comment:
I think it would be a good idea to keep these callbacks wired up, to prevent a major breaking change. Some emulator cores (including my own) use this pattern. I don't see any compelling reason to remove these while making the other changes, unless I'm missing something?

Reply:
I think I'll end up rolling this back in conjunction with the changes to `RetroInstance::on_unload_game`. But, to clarify, are you saying you'd like to keep the previous approach of initializing the core in `RetroCore::init`?

Reply:
Just having the callbacks wired up is all I mean. It would be very surprising to grab the crate, read some libretro docs only to find that `retro_init()` and `retro_deinit()` aren't exposed. That's all this glue layer is meant to be, expose the API calls that a front-end expects, and let the `RetroInstance` code "rustify" them.

Reply:
I guess that's true, but without `RetroCore::init` there's nothing for `RetroInstance::on_init` to do. The old implementation simply called `RetroCore::init` to initialize the `system` field, but I removed `RetroCore::init` and the assignment to `system` is now happening in `RetroInstance::on_load_game`. It seems strange to leave a no-op method in `RetroInstance`.

Would you like me to re-add `RetroCore::init` as a function (rather than a method) with a default no-op implementation in case the user wants to be able to observe that `retro_init` was called? If so, should I add a `RetroCore::deinit` function to go with it?